src/devices/bus/coco/coco_fdc.cpp

```diff
@@ -2,115 +2,121 @@
+	The CoCo uses either the WD1793 or the WD1773, the Dragon uses the WD2797,
+	which mostly uses the same command set with some subtle differences, most
+	notably the 2797 handles disk side select internally. The Dragon Alpha also
+	uses the WD2797, however as this is a built in interface and not an external
+	cartridge, it is dealt with in the main coco.cpp file.
+
+	The wd's variables are mapped to $FF48-$FF4B on the CoCo and on $FF40-$FF43
+	on the Dragon.	In addition, there is another register
+	called DSKREG that controls the interface with the wd1793.	DSKREG is
+	detailed below:
+
+	References:
+		CoCo:	Disk Basic Unravelled
+		Dragon: Inferences from the PC-Dragon source code
+				DragonDos Controller, Disk and File Formats by Graham E Kinns
+
+	---------------------------------------------------------------------------
+
+	DSKREG - the control register
+	CoCo ($FF40)
+
+	Bit
+	7 halt enable flag
+	6 drive select #3
+	5 density (0=single, 1=double) and NMI enable flag
+	4 write precompensation
+	3 drive motor activation
+	2 drive select #2
+	1 drive select #1
+	0 drive select #0
+
+	Reading from $FF48-$FF4F clears bit 7 of DSKREG ($FF40)
+
+	---------------------------------------------------------------------------
+
+	Disto Super Controller II
+		Mini Expansion Bus
+		No Halt Extension. Implemented by using a read and write cache.
+
+			CachDat - Cache Data Register
+				$FF74 & $FF75: Read/Write cache data.
+
+				Normally cache size is 256 bytes. But a common upgrade is to change
+				it to 512 bytes.
+
+			CachCtrl - Cache Controller
+				$FF76 & $FF77:
+					write
+						clears buffer counter
+
+					read bit:
+						7:  0 - indicates an interrupt request from the disk controller
+							1 - no interrupt
+						3:  0 - disable CART / FIRQ (masked)
+							1 - enable CART / FIRQ
+						2:  0 - normal NMI
+							1 - masked NMI
+						1:  0 - compatible I/O mode
+							1 - buffered I/O mode
+						0:  0 - buffered write operation
+							1 - buffered read operation
+
+		There was a jumper to set the addresses to be FF58 to FF5B. Not implemented.
 
 *********************************************************************/
 
 #include "emu.h"
-#include "cococart.h"
 #include "coco_fdc.h"
-#include "imagedev/floppy.h"
-#include "machine/msm6242.h"
-#include "machine/ds1315.h"
-#include "machine/wd_fdc.h"
-#include "machine/ram.h"
+#include "cococart.h"
 #include "formats/dmk_dsk.h"
+#include "formats/fs_coco_os9.h"
+#include "formats/fs_coco_rsdos.h"
 #include "formats/jvc_dsk.h"
-#include "formats/vdk_dsk.h"
-#include "formats/sdf_dsk.h"
 #include "formats/os9_dsk.h"
-#include "formats/fs_coco_rsdos.h"
-#include "formats/fs_coco_os9.h"
+#include "formats/sdf_dsk.h"
+#include "formats/vdk_dsk.h"
+#include "imagedev/floppy.h"
+#include "machine/ds1315.h"
+#include "machine/input_merger.h"
+#include "machine/msm6242.h"
+#include "machine/wd_fdc.h"
+#include "meb_intrf.h"
```

Comment:
Please put this back in order:
* Prefix header
* Module header
* Other local headers
* Libraries from most dependent to least dependent (i.e. stuff from devices comes before stuff from lib/formats)

Reply:
Fixed.